### ISSUE TITLE
Minor issues that keep annoying me

### ISSUE THREAD

**Issue by Drenderer** [Drenderer/klax#70]
## Bad variable names
Some variable names in klax are currently badly chosen:
- [x] The loss Protocol requires `data`. That really should be called `batch`. To retain backwards compatibility, we should consider relaxing the Loss Protocol and only require positional arguments for the loss functions. Then the user can call the arguments whatever fits best.
- [x] All over the place, `batch_axis` is used, which should be called `batch_axes`. Besides being grammatically correct in most cases where one actually specifies a `batch_axes`, this would make it consistent with `jax.vmap(..., in_axes)` and many more...
- [ ] ...

## Typing
There are some basic typing oversights:
- [x] `klax.finalize` is missing an output type. Without having put much thought into this, I fear that a generic type will not be 100% correct:
  ```python
  def finalize[T: PyTree](tree: T) -> T: ...
  ```
  While this should work perfectly fine for many use cases, we should consider possible edge cases where it might be incorrect.
- [ ] `klax.split_data` does not have a generic type, but uses `PyTree`, meaning it uses `Any`.
- [ ] The already mentioned Loss protocol is overly restrictive. It does not allow for the loss function arguments to be changed, and it does not allow applying custom types to the arguments. 
- [x] `klax.fit` demands an `eqx.Module` as input type, but in practice it could be any pytree, e.g., a tuple of modules. Therefore, the type annotation should be changes to a generic pytree.

## Other
- [x] The dependency on Paramax is unnecessary (?)

**Comment by Drenderer** [comment]
Since the branch has not yet been merged, I reopened this issue to collect any remaining annoyances.
